debezium-core/src/main/java/io/debezium/transforms/tracing/ActivateOpenTracingSpan.java

```diff
@@ -0,0 +1,159 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivateOpenTracingSpan.class);
+
+    private static final String DEFAULT_OPENTRACING_SPAN_VALUE = "opentracingspan";
+    private static final String DEFAULT_OPENTRACING_OPERATION_NAME = "debezium";
+
+    private static final String TAG_PREFIX = "debezium.";
+
+    private static final Field OPENTRACING_SPAN_FIELD = Field.create("opentracing.span.field")
+            .withDisplayName("OpenTracing Span Field")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDefault(DEFAULT_OPENTRACING_SPAN_VALUE)
+            .withDescription("The name of the field notaining java.util.Properties representation of OpenTracing span.");
+
+    private static final Field OPENTRACING_OOPERATION_NAME = Field.create("opentracing.operation.name")
+            .withDisplayName("OpenTracing Operation Name")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDefault(DEFAULT_OPENTRACING_OPERATION_NAME)
+            .withDescription("The name of the field notaining java.util.Properties representation of OpenTracing span.");
+
+    private String spanField;
+    private String operationName;
+
+    private SmtManager<R> smtManager;
+
+    @Override
+    public void configure(Map<String, ?> props) {
+        Configuration config = Configuration.from(props);
+        final Field.Set configFields = Field.setOf(OPENTRACING_SPAN_FIELD, OPENTRACING_OOPERATION_NAME);
+
+        if (!config.validateAndRecord(configFields, LOGGER::error)) {
+            throw new ConnectException("Unable to validate config.");
+        }
+
+        spanField = config.getString(OPENTRACING_SPAN_FIELD);
+        operationName = config.getString(OPENTRACING_OOPERATION_NAME);
+
+        smtManager = new SmtManager<>(config);
+    }
+
+    @Override
+    public R apply(R record) {
+        final Tracer tracer = GlobalTracer.get();
+        if (tracer == null) {
+            return record;
+        }
+
+        // In case of tombstones or non-CDC events (heartbeats, schema change events),
+        // leave the value as-is
+        if (record.value() == null || !smtManager.isValidEnvelope(record)) {
+            return record;
+        }
+
+        final Struct envelope = (Struct) record.value();
+        final Struct after = envelope.getStruct(Envelope.FieldName.AFTER);
+        final Struct source = envelope.getStruct(Envelope.FieldName.SOURCE);
+
+        final SpanBuilder spanBuilder = tracer.buildSpan(operationName);
+        addFieldToSpan(spanBuilder, envelope, Envelope.FieldName.OPERATION, TAG_PREFIX);
+        addFieldToSpan(spanBuilder, envelope, Envelope.FieldName.TIMESTAMP, TAG_PREFIX);
+
+        if (source != null) {
+            final String sourcePrefix = TAG_PREFIX + "source.";
```

Comment:
`source` describes the postion in the transaction log where the change was recorded. Debezium itself uses field named `source` in its messages for that puporse so I thought it might fit to align the names.

Reply:
Alright, I thought it was referring to an external system.

Reply:
In a way, it does. eg. `source.gtid` refers to the transaction id of the event in the source database where this change event is orginating from. It does _not_ refer to the system that triggered the change in the database originally.